### ISSUE TITLE
[BUG]MetricsTracker multiplies `accelerate.num_processes` twice, causing `samples/epochs` to grow by n² in multi-process training

### ISSUE THREAD

**Issue by Daniel-rmc** [huggingface/lerobot#2579]
## Description
When training with Accelerate in multi-process mode (`accelerate.num_processes > 1`), the logged `smpl` (samples) and `epch` (epochs) increase faster than expected by a factor of `num_processes`, effectively resulting in 4× on 2 GPUs or `n²` on `n` GPUs. This skews monitoring on platforms like Weights & Biases (wandb), where the inferred effective batch size no longer matches the configured value.

Example logs:
```
INFO ... Effective batch size: 32 x 2 = 64
INFO ... step:200 smpl:6K ep:26 epch:0.12 ...
```
In practice, each step increments `samples` by `(batch_size_per_device * num_processes) * num_processes`, i.e., `num_processes` is applied twice.

## Affected code
- `src/lerobot/scripts/lerobot_train.py`
  - `MetricsTracker` is constructed with the effective (global) batch size:
    ```python
    effective_batch_size = cfg.batch_size * accelerator.num_processes
    train_tracker = MetricsTracker(
        effective_batch_size,
        dataset.num_frames,
        dataset.num_episodes,
        train_metrics,
        initial_step=step,
        accelerator=accelerator,
    )
    ```
- `src/lerobot/utils/logging_utils.py`
  - `MetricsTracker.step()` multiplies by `num_processes` again:
    ```python
    self.samples += self._batch_size * (self.accelerator.num_processes if self.accelerator else 1)
    ```
  - `MetricsTracker.__init__` initial samples do not consider `num_processes`:
    ```python
    self.samples = self.steps * self._batch_size
    ```

## Reproduction steps
1. Launch multi-process training with Accelerate (e.g., 2 GPUs, `num_processes=2`) and set `cfg.batch_size=32`.
2. Logs will show `Effective batch size: 32 x 2 = 64`.
3. During training, `smpl/epch` increase per step by `(64 * 2)` instead of `64`, yielding a 4× inflation; with 4 GPUs this becomes 16×.
4. On wandb, the batch size inferred from `samples/epochs` appears 4× larger than the configured per-device batch.

## Expected behavior
- The global `samples` should increase per step by `batch_size_per_device * num_processes` (apply num_processes once), or by the provided global batch if that is what's passed in — without multiplying by `num_processes` again.
- Initial `samples` (when resuming with `initial_step > 0`) should reflect the global sample count consistently.

## Actual behavior
- `train.py` applies `num_processes` to compute `effective_batch_size`, and `MetricsTracker.step()` applies `num_processes` once more, resulting in `batch_size_per_device * (num_processes^2)` per step.
- In `__init__`, `samples = steps * _batch_size` does not multiply by `num_processes`, so resuming from a non-zero step undercounts the initial samples.

## Root cause
- num_processes (`num_processes`) is applied in two places:
  - In `train.py` when building `MetricsTracker` with `effective_batch_size`.
  - In `MetricsTracker.step()` where `* accelerator.num_processes` is used.
- Initial sample count in `__init__` omits num_processes, making resume inaccurate.

## Proposed fixes (choose one)
- Option A (recommended): Pass per-device batch size to `MetricsTracker` and let it account for num_processes internally.
  - In `lerobot_train.py`:
    ```python
    train_tracker = MetricsTracker(
        cfg.batch_size,  # per-device batch size
        dataset.num_frames,
        dataset.num_episodes,
        train_metrics,
        initial_step=step,
        accelerator=accelerator,
    )
    ```
  - In `logging_utils.py`:
    ```python
    # __init__
    world_size = self.accelerator.num_processes if self.accelerator else 1
    self.samples = self.steps * self._batch_size * world_size

    # step()
    self.samples += self._batch_size * world_size
    ```

- Option B: Keep passing global batch size and remove world-size multiplication from `MetricsTracker`.
  - In `lerobot_train.py`, keep `effective_batch_size`.
  - In `logging_utils.py`:
    ```python
    # __init__ stays
    self.samples = self.steps * self._batch_size

    # step(): do not multiply by num_processes
    self.samples += self._batch_size
    ```
  - Note: Under this option, all callers of `MetricsTracker` must consistently pass the global batch size.

## Validation (example)
- Let `batch_size_per_device=32`, `num_processes=2`, `initial_step=10`.
- Option A:
  - Init: `samples = 10 * 32 * 2 = 640`
  - One `step()`: `samples += 32 * 2 = 64` → 704
- Option B (pass `effective_batch_size=64`):
  - Init: `samples = 10 * 64 = 640`
  - One `step()`: `samples += 64` → 704
- Both agree and avoid the 4× inflation.

## Impact
- After fixing, wandb `samples/epochs` align with configured batch size and num_processes.
- `epochs = samples / num_frames` reflects accurate training progress.

## Environment
- OS: Linux (Ubuntu 24.04.2 LTS)
- Python: 3.10
- Framework: `accelerate` (`DistributedDataParallelKwargs(find_unused_parameters=True)`)
- Training script: `src/lerobot/scripts/lerobot_train.py`
- Module affected: `src/lerobot/utils/logging_utils.py`

## Additional suggestions
- Log and distinguish:
  - per-device batch size (config)
  - num_processes (`accelerate.num_processes`)
  - global batch size (product of the above)
- Record all three in wandb config for clearer comparisons across runs.

**Comment by Daniel-rmc** [comment]
Please add a bug label to this issue to help attract attention and get it fixed faster.
